app/assets/javascripts/common/admins.js

```diff
@@ -0,0 +1,213 @@
+
+      this.updateChildrenCheckedState(targetCheckbox, ACCESS_LIST_INPUT_SELECTOR)
+      this.updateParentCheckedState(targetCheckbox, ACCESS_LIST_INPUT_SELECTOR)
+    })
+  },
+
+  resourceRowCollapseListener: function () {
+    const collapsibleItems = [
+      this.facilityAccessItemsPadding(),
+      this.facilityAccessItemsAccessRatio()
+    ].map(htmlCollection => Array.from(htmlCollection)).flat()
+
+    for (const item of collapsibleItems) {
+      item.addEventListener("click", this.onFacilityAccessItemToggled.bind(this))
+    }
+  },
+
+  toggleAccessTreeVisibility: function (isPowerUser) {
+    if (isPowerUser) {
+      this.facilityAccess().classList.add("hidden")
+      this.facilityAccessPowerUser().classList.remove("hidden")
+    } else {
+      this.facilityAccess().classList.remove("hidden")
+      this.facilityAccessPowerUser().classList.add("hidden")
+    }
+  },
+
+  onAccessLevelChanged: function ({ target }) {
+    this.toggleAccessTreeVisibility(target.value === ACCESS_LEVEL_POWER_USER)
+  },
+
+  toggleItemCollapsed: function (element) {
+    const collapsed = element.classList.contains("collapsed")
+
+    if (collapsed) {
+      element.classList.remove("collapsed")
+    } else {
+      element.classList.add("collapsed")
+    }
+  },
+
+  onFacilityAccessItemToggled: function ({ target }) {
+    const children = Array.from(target.closest("li").childNodes)
+    const parentItem = target.closest(".access-item")
+    const wrapper = children.find(containsClass("access-item-wrapper"))
+    if (wrapper) {
+      this.toggleItemCollapsed(parentItem)
+    }
+  },
+
+  updateParentCheckedState: function (element, selector) {
+    // find parent and sibling checkboxes
+    const parent = (element.closest(["ul"]).parentNode).querySelector(selector)
+    const siblings = nodeListToArray(selector, parent.closest("li").querySelector(["ul"]))
+
+    // get checked state of siblings
+    // are every or some siblings checked (using Boolean as test function)
+    const checkStatus = siblings.map(check => check.checked)
+    const every = checkStatus.every(Boolean)
+    const some = checkStatus.some(Boolean)
+
+    // check parent if all siblings are checked
+    // set indeterminate if not all and not none are checked
+    parent.checked = every
+    parent.indeterminate = some && !every
+
+    // recurse until check is the top most parent
+    if (element !== parent) this.updateParentCheckedState(parent, selector)
+  },
+
+  updateChildrenCheckedState: function (parent, selector) {
+    // check/uncheck children (includes check itself)
+    const children = nodeListToArray(selector, parent.closest("li"))
+
+    children.forEach(child => {
+      // reset indeterminate state for children
+      child.indeterminate = false
+      child.checked = parent.checked
+    })
+  },
+
+  onAsyncLoaded: function () {
+    const _self = this
+    document.addEventListener('render_async_load', function (_event) {
```

Comment:
1/2:

This is one of the main changes in this file, we're loading the access tree related JS after a callback hook from render_async_load, once it has finished loading.